From 679c9fd8c7aa976c3f09f3b4e44125a52ce550a8 Mon Sep 17 00:00:00 2001
From: Michal Schmidt <mschmidt@redhat.com>
Date: Fri, 5 Sep 2014 19:49:58 -0400
Subject: [ethernet] bnx2x: fix tunneled GSO over IPv6

Message-id: <1409946615-5862-54-git-send-email-mschmidt@redhat.com>
Patchwork-id: 91612
O-Subject: [RHEL7.1 PATCH 53/70] bnx2x: fix tunneled GSO over IPv6
Bugzilla: 1119861 1060671 1089402
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>

BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1060671
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1089402
BZ: https://bugzilla.redhat.com/show_bug.cgi?id=1119861

Upstream commit:
commit 05f8461bf746660ea725f922cfae9af66be1f79f
Author: Dmitry Kravkov <Dmitry.Kravkov@qlogic.com>
Date:   Thu Aug 28 16:54:24 2014 +0300

    bnx2x: fix tunneled GSO over IPv6

    Set correct bit for packed description.

    Introduced in e42780b66aab88d3a82b6087bcd6095b90eecde7
        bnx2x: Utilize FW 7.10.51

    Reported-by: Dan Carpenter <dan.carpenter@oracle.com>
    Signed-off-by: Dmitry Kravkov <Dmitry.Kravkov@qlogic.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
index 12430b0..58410ea 100644
--- a/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
+++ b/drivers/net/ethernet/broadcom/bnx2x/bnx2x_cmn.c
@@ -3640,7 +3640,7 @@ static void bnx2x_update_pbds_gso_enc(struct sk_buff *skb,
   pbd2->fw_ip_hdr_to_payload_w =
    hlen_w - ((sizeof(struct ipv6hdr)) >> 1);
   pbd_e2->data.tunnel_data.flags |=
-   1 /*IPv6*/ << ETH_TUNNEL_DATA_IP_HDR_TYPE_OUTER;
+   ETH_TUNNEL_DATA_IP_HDR_TYPE_OUTER;
  }
 
  pbd2->tcp_send_seq = bswab32(inner_tcp_hdr(skb)->seq);
-- 
1.7.1